rfd/0047-drop-vendor.md

```diff
@@ -0,0 +1,144 @@
+---
+authors: Zac Bergquist <zac@goteleport.com
+state: draft
+---
+
+# RFD 47 - Drop the vendor directory
+
+## What
+
+This RFD proposes that we stop committing the `vendor/` directory to source control.
+This is a developer-facing change only and has no user-visible impact.
+
+The `vendor/` directory contains the Go source code for the third party dependencies
+required to build Teleport, and exists largely as a relic of early Go (pre-modules).
+
+Prior to Go 1.11's introduction of
+[modules](https://go.dev/blog/using-go-modules) (2018), vendoring was the
+commonly-accepted convention for maintaining a set of project-specific
+dependencies, ensuring that a vendored copy of a dependency is preferred over
+a `$GOPATH`-wide version.
+
+Teleport existed before Go modules, and has leveraged vendoring for several
+years. When Teleport was converted to a Go module, we adopted the
+[`go mod vendor`](https://go.dev/ref/mod#go-mod-vendor) tool as a way to leverage
+modules while keeping the vendor directory and workflow in-tact.
+
+## Why
+
+At a high level, we propose eliminating the vendor directory for the following reasons:
+
+1. It is no longer necessary, and the community at large is moving away from vendoring.
+1. Dropping the vendor directory would result in a better developer experience.
+
+## Details
+
+### No Longer Necessary
+
+In one of the [original proposals](https://research.swtch.com/vgo-module) for Go
+modules, Russ Cox writes:
+
+>    We want to eliminate vendor directories. They were introduced for
+>    reproducibility and availability, but we now have better mechanisms.
+>    Reproducibility is handled by proper versioning, and availability is handled by
+>    caching proxies.
+
+In short, vendoring was a solution for two common problems:
+
+1. Reproducible builds
+1. Availability of depdencies
```

Comment:
```suggestion
2. Availability of dependencies
```

Reply:
I typically leave everything numbered `1.` since Github renders the ordered list correctly, but this is short enough, sure.